
# HG changeset patch
# User Timo Sirainen <tss@iki.fi>
# Date 1384951064 -7200
# Node ID f4eb4b5884b2e89e146a32b2d1db1f14f753008f
# Parent  4a0c03f98ca350634430e5c3bd5f8e5efc030a45
lib-storage: mail_get_*header*() didn't remove leading whitespace for headers from cache.
But it was removed when the headers were found by parsing the full header.

diff -r 4a0c03f98ca3 -r f4eb4b5884b2 src/lib-storage/index/index-mail-headers.c
--- a/src/lib-storage/index/index-mail-headers.c	Tue Nov 19 23:18:11 2013 +0200
+++ b/src/lib-storage/index/index-mail-headers.c	Wed Nov 20 14:37:44 2013 +0200
@@ -653,7 +653,8 @@
 	/* cached. skip "header name: " parts in dest. */
 	for (i = 0; i < len; i++) {
 		if (data[i] == ':') {
-			if (i+1 != len && data[++i] == ' ') i++;
+			while (IS_LWSP(data[i+1])) i++;
+			if (i+1 != len) i++;
 
 			/* @UNSAFE */
 			len = get_header_size(dest, i);

